indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/SinglePhaseSubTaskSpec.java

```diff
@@ -66,4 +66,28 @@ public SinglePhaseSubTask newSubTask(int numAttempts)
         new DummyForInjectionAppenderatorsManager()
     );
   }
+
+  @Override
+  public SinglePhaseSubTask newSubTaskWithBackwardCompatibleType(int numAttempts)
+  {
+    return new SinglePhaseSubTask(
+        null,
+        getGroupId(),
+        null,
+        getSupervisorTaskId(),
+        numAttempts,
+        getIngestionSpec(),
+        getContext(),
+        null,
+        null,
+        new DummyForInjectionAppenderatorsManager()
+    )
+    {
+      @Override
+      public String getType()
+      {
+        return SinglePhaseSubTask.OLD_TYPE_NAME;
```

Comment:
Will this really affect how the task is serialized? I thought the `type` field would end up based solely on whatever Jackson thinks the type code for that class is, based on `@JsonTypeName` or `@JsonSubTypes` annotations.

Could you include a test that makes sure it serializes correctly? (Maybe serialize it, then deserialize as a Map and check the `type` field.)

Reply:
Returning a different `type` works, but the class should be registered on Jackson properly. I fixed it and added a unit test.